src/jit/gentree.h

```diff
@@ -4097,6 +4101,30 @@ inline bool GenTree::IsZero()
     return false;
 }
 
+// Note that, unlike IsZero(), this only returns true only for integral types.
+inline bool GenTree::IsIOne()
+{
+    if ((gtOper == GT_CNS_INT) && (gtIntConCommon.IconValue() == 1))
+        return true;
+
+    if ((gtOper == GT_CNS_LNG) && (gtIntConCommon.LngValue() == 1))
+        return true;
+
+    return false;
+}
+
+// Note that, unlike IsZero(), this only returns true only for integral types.
+inline bool GenTree::IsIMinusOne()
```

Comment:
As I said above, we should either rename this to IsIntegralMinusOne() or handle floating-point constants as well.